lib/countly.js

```diff
@@ -272,1928 +220,2506 @@
+                var ca = document.cookie.split(";");
+                for (var i = 0, max = ca.length; i < max; i++) {
+                    var c = ca[i];
+                    while (c.charAt(0) === " ") {
+                        c = c.substring(1, c.length);
+                    }
+                    if (c.indexOf(nameEQ) === 0) {
+                        return c.substring(nameEQ.length, c.length);
+                    }
+                }
+                return null;
             }
-            store("cly_event", eventQueue);
         }
 
-        //process request queue with event queue
-        if (!offlineMode && requestQueue.length > 0 && readyToProcess && getTimestamp() > failTimeout) {
-            readyToProcess = false;
-            var params = requestQueue[0];
-            //check if any consent to sync
-            if (hasAnyProperties(syncConsents)) {
-                if (consentTimer) {
-                    clearTimeout(consentTimer);
-                    consentTimer = null;
+        /**
+         *  Migrate from old storage to new app_key prefixed storage
+         */
+        function migrate() {
+            if (store(self.namespace + "cly_id", undefined, false, true)) {
+                //old data exists, we should migrate it
+                store("cly_id", store(self.namespace + "cly_id", undefined, false, true));
+                store("cly_event", store(self.namespace + "cly_event", undefined, false, true));
+                store("cly_session", store(self.namespace + "cly_session", undefined, false, true));
+
+                //filter out requests with correct app_key
+                var requests = store(self.namespace + "cly_queue", undefined, false, true);
+                if (Array.isArray(requests)) {
+                    requests = requests.filter(function(req) {
+                        return req.app_key === self.app_key;
+                    });
+                    store("cly_queue", requests);
                 }
-                params.consent = JSON.stringify(syncConsents);
-                syncConsents = {};
-            }
-            log("Processing request", params);
-            store("cly_queue", requestQueue, true);
-            sendXmlHttpRequest(Countly.url + apiPath, params, function(err, params){
-                log("Request Finished", params, err);
-                if (err) {
-                    failTimeout = getTimestamp() + failTimeoutAmount;
+                if (store(self.namespace + "cly_cmp_id", undefined, false, true)) {
+                    store("cly_cmp_id", store(self.namespace + "cly_cmp_id", undefined, false, true));
+                    store("cly_cmp_uid", store(self.namespace + "cly_cmp_uid", undefined, false, true));
                 }
-                else {
-                    //remove first item from queue
-                    requestQueue.shift();
+                if (store(self.namespace + "cly_ignore", undefined, false, true)) {
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/manual/ar2rsawseen/countly-sdk-web/pullRequest?prid=6356797)